packages/css/src/index.ts

```diff
@@ -268,8 +295,12 @@ export const css = (args: ThemeUIStyleObject = {}) => (
     const val = typeof x === 'function' ? x(theme) : x
 
     if (key === 'variant') {
-      const variant = css(get(theme, val))(theme)
-      result = { ...result, ...variant }
+      const styles = css(
+        merge.all(
+          ...val.split(VARIANT_SEPARATOR).map(variants => get(theme, variants))
+        )
+      )(theme)
```

Comment:
This is where all the magic happens.

And by magic, I mean `string.split` and `deepmerge` without much else.